electrumsv/daemon.py

```diff
@@ -146,14 +150,55 @@ def __init__(self, fd, is_gui: bool) -> None:
             app_state.fx = FxTask(app_state.config, self.network)
             self.fx_task = app_state.async_.spawn(app_state.fx.refresh_loop)
         self.wallets = {}
-        # Setup JSONRPC server
+
+        # RPC API - (synchronous) - self.run()
         self.init_server(config, fd, is_gui)
-        # self.init_thread_watcher()
+
+        # REST API - (asynchronous) - async_ loop/thread
+        self.init_restapi_server(config, fd)
+        self.default_endpoints = {"/"    : self.status,
+                                  "/ping": self.rest_ping}
+        app_state.dapp_extensions_added = False
+        self.configure_restapi_server(self.default_endpoints)
+
+    # ----- Default External API ----- #
+
+    async def status(self, request):
+        return web.json_response({
+            "status": "success",
+        })
+
+    async def rest_ping(self, request):
+        return web.json_response({
+            "value": "pong"
+        })
+
+    # -------------------------------- #
+
+    def configure_restapi_server(self, extension_endpoints: Dict[str, Any]):
+        self.rest_server.register_new_endpoints(extension_endpoints)
+        self.logger.debug(f"added default rest api endpoints: {list(extension_endpoints.keys())}")
+        pass
+
+    def init_restapi_server(self, config: SimpleConfig, fd) -> None:
+        host = config.get('rpchost', '127.0.0.1')
+        port = 9999  # hard-code until added to config
+
+        # Basic Auth not yet configured. Credentials shared with rpc currently.
+        username, password = get_rpc_credentials(config, is_restapi=True)
+        try:
+            self.rest_server = AiohttpServer(host=host, port=port, username=username,
+                password=password, extension_endpoints=None)
+
+        except Exception as e:
```

Comment:
Can you put the explicit exceptions that are failing here instead of Exception?

Reply:
I don't really know what exceptions could be raised when trying to instantiate the aiohttp server... could be any number of things. What should I do in that case?

Reply:
I would say that we should move away from catching these exceptions, and should just let them raise. There should be a line where we say, if this has an exception then the user should report a bug and we'll add handling for that exception. And the other side of the line is that we have no option but to catch-all Exception. All except statements should (although we've been slack on this) have a comment qualifying what they catch and why they catch it, so that if someone were to come along and change it from catching an exception to proper handling for the given case, they would know they were safe in removing the exception handling.

Reply:
Okay. Sounds fair enough. I have removed the try / except to let it raise.